slf4j-api/src/main/java/org/slf4j/helpers/BasicMDCAdapter.java

```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2004-2011 QOS.ch
+ * Copyright (c) 2004-2011 QOS.ch, Copyright (C) 2015 Google Inc.
```

Comment:
Everyone who writes code, owns the copyright and is the author. When someone gives you a patch, they are just *licensing* it to the project maintainer, not transferring ownership.

In general many opensource projects have a contributor licensing agreement (CLA) which details the license under which the terms that the intellectual property has been contributed. It doesn't seem as though SLF4J has such a CLA which stipulates the terms.

More details are here:
http://en.wikipedia.org/wiki/Contributor_License_Agreement

Here are some example CLA agreements for projects which receive many individual contributions:
http://www.apache.org/licenses/icla.txt
https://www.python.org/psf/contrib/contrib-form/
http://www.eclipse.org/legal/CLA.php
https://docs.google.com/a/google.com/forms/d/13WR8m5uZ2nAkJH41k7GdVBXAAbzDk00vxtEYjd6Imzg/viewform?formkey=dDViT2xzUHAwRkI3X3k5Z0lQM091OGc6MQ

Many projects have a file which details the terms for contributing to them, for example:
https://github.com/apache/ant/blob/master/CONTRIBUTING.md which stipulates that a CLA must by signed unless the patch is trivial.